backend/main.py

```
@@ -70,101 +73,21 @@ async def rate_limit_middleware(request: Request, call_next):
-                model="gpt-4o-mini-tts",
-                voice="ash",
-                input=GREETING_TEXT,
-                instructions=CARL_SYSTEM_PROMPT,
-                response_format="mp3"
-            )
-            audio_bytes = response.content
+            audio_bytes = await agent.generate_greeting(GREETING_TEXT)
             return StreamingResponse(iter([audio_bytes]), media_type="audio/mpeg")
         except Exception as e:
             print(f"Error generating greeting audio: {e}")
             raise HTTPException(status_code=500, detail="Failed to generate greeting audio.")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)